docs/data/data-grid/row-definition/row-definition.md

```diff
@@ -30,7 +30,11 @@ If the row's identifier is not called `id`, then you need to use the `getRowId`
 The following demo shows how to use `getRowId` to grab the unique identifier from a property named `internalId`:
 
 ```tsx
-<DataGrid getRowId={(row) => row.internalId} />
+function getRowId(row) {
+  return row.internalId;
+}
+
+<DataGrid getRowId={getRowId} />;
```

Comment:
I've been thinking about this and I'm not sure which notation to use in the docs.

The first one is more compact & readable, but the second one is the one people should actually use. And this applies to all props passed to the grid. I'm wondering if we shouldn't put a big statement in the intro about this and leave everything as the compact notation.

Because the warning below is basically repeating what the [perf section](https://mui.com/x/react-data-grid/performance/#extract-static-objects-and-memoize-root-props) says. Wdyt?

cc @cherniavskii

Reply:
> the second one is the one people should actually use

I think we should use this one.

> I'm wondering if we shouldn't put a big statement in the intro about this and leave everything as the compact notation.

A single place for this warning is not enough, since there are countless docs entry points.
I think duplicating this info is way better. Maybe extract a reusable warning component to use across the docs?